pkg/util/azureclient/mgmt/insights/activitylogs.go

```diff
@@ -4,25 +4,38 @@ package insights
 // Licensed under the Apache License 2.0.
 
 import (
-	mgmtinsights "github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-03-01/insights"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
+
+	//"github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-03-01/insights"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/insights/insights"
 )
 
+/* TODO (dranders): Change back to the normal azure-sdk-for-go
+once https://github.com/Azure/azure-sdk-for-go/issues/14704 is completed, or the 2015-04-01 azure API version
+successfully finds the missing objects:
+
+	activityLog.OperationName.Value == "Microsoft.Compute/virtualMachines/redeploy/action"
+	https://github.com/Azure/ARO-RP/blob/master/test/e2e/adminapi_redeployvm.go#L64-L70
+
+Last information on a fix to be provided by Azure stated sometime in 2nd half of calendar year 2021
+More details in the github issue, and the referenced ICM (see github issue for icm id)
+*/
+
 // ActivityLogsClient is a minimal interface for azure ActivityLogsClient
 type ActivityLogsClient interface {
 	ActivityLogsClientAddons
 }
 
 type activityLogsClient struct {
-	mgmtinsights.ActivityLogsClient
+	insights.ActivityLogsClient
```

Comment:
Can we import it as `mgmtinsights` so we can just swap the import back to the SDK once we upgrade to a version which contains a new package?

Applies to other files too.

Reply:
The include is correct with this import name, for both original SDK version and for the modified version. In fact, they are both in the `insights` namespace. This fixes a future error with later golang versions.

Reply:
Not sure that I understand. Probably my initial comment was unclear. I'll try to explain a bit better.

Currently in `master` we import `"github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-03-01/insights"` package use `mgmtinsights` alias for it in this (and some other files) for it.

In this PR we replace the above package with `"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/insights/insights"` which should be a drop-in replacement. So in this file (and in other similar files) I would only expect imprt to change and nothing else in the code:

```diff
- mgmtinsights "github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-03-01/insights" 
+ mgmtinsights "github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/insights/insights"
```

Reply:
@m1kola 

![image](https://user-images.githubusercontent.com/5228837/121877871-60f5fc00-cd0b-11eb-8fec-5cd0ccd13d1d.png)

Reply:
`pkg/util/azureclient/mgmt/insights/activitylogs.go: github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/insights/insights is imported as "mgmtinsights", should be [""]`

Reply:
Tests fail with original namespace. Original namespace was not required in the first place.

Reply:
@drewandersonnz  Right. `mgmtinsights` is enforced by this `validate-imports` rule:

https://github.com/Azure/ARO-RP/blob/8db01ffbc3943abb2054d35bad114752ec6b0816/hack/validate-imports/validate-imports.go#L75-L78

We can modify `validate-imports` to allow us to import the temporary package as `mgmtinsights` as well, but I think it is not worth the effort (will introduce even more diff in this PR).

Let's leave it without alias.